### PR TITLE
Updated APIs in the provided started code to match Swift 3 style

### PR DESCRIPTION
**Pull request by brockboland**
I also re-indented all the changed files, since a bunch of them were way off.

@vokal/ios-developers code review please?

### CODE REVIEW

**Comment by bryanluby** [comment]
Sure

**Comment by brockboland** [comment]
@bryanluby improved the naming of some methods, and also dropped our Swift 3 templates for mogenerator: the ones in mogen are very similar but cover some edge cases ours didn't.

**Comment by brockboland** [comment]
@bryanluby ok NOW all my changes are here. Sorry about that.

File: Vokal-Swift.xctemplate/TokenStorageHelper.swift [hunk 1]
```diff
@@ -24,12 +24,12 @@ struct TokenStorageHelper {
...
-    - parameter email:     The user's email address.
-    - parameter authToken: The current auth token.
-    */
-    static func storeAuthorizationTokenForUserEmail(_ email: String, authToken: String) {
+     Stores the authorization token for a given user's email in the keychain.
+     
+     - parameter email:     The user's email address.
+     - parameter authToken: The current auth token.
+     */
+    static func store(authorizationToken: String, for email: String) {
```
Comment (bryanluby) [inline Vokal-Swift.xctemplate/TokenStorageHelper.swift]: How about `forEmail`

File: Vokal-Swift.xctemplate/ResetPasswordAPI.swift [hunk 2]
```diff
@@ -31,66 +31,66 @@ struct ResetPasswordAPI {
...
+     Requests a password reset for a given email address. If the email exists, this will trigger
+     an email to the user with a reset request code which they can then use to reset their
+     password. If the email doesn't exist, the server just smiles and nods to avoid disclosing
+     what emails are actually related to accounts.
+     
+     - parameter email:      The email the user is requesting a password reset for.
+     - parameter success:     The closure to execute if the request succeeds.
+     - parameter failure:     The closure to execute if the request fails.
+     */
+    static func requestPasswordReset(for email: String,
```
Comment (bryanluby) [inline Vokal-Swift.xctemplate/ResetPasswordAPI.swift]: How about `forEmail`

File: Vokal-Swift.xctemplate/ResetPasswordAPI.swift [hunk 3]
```diff
@@ -31,66 +31,66 @@ struct ResetPasswordAPI {
...
-                    success: success,
-                    failure:failure)
+     Resets the password for a user who has received a password reset code via email.
+     
+     - parameter code:            The reset password code provided by the user.
+     - parameter updatedPassword: The password the user now wants to use.
+     - parameter success:     The closure to execute if the request succeeds.
+     - parameter failure:     The closure to execute if the request fails.
+     */
+    static func resetPassword(with code: String,
```
Comment (bryanluby) [inline Vokal-Swift.xctemplate/ResetPasswordAPI.swift]: How about `withCode`

File: Vokal-Swift.xctemplate/UserAPITests.swift [hunk 4]
```diff
@@ -42,31 +42,31 @@ class UserAPITests: XCTestCase {
...
-                }
-                
-                expectation.fulfill()
-            },
-            failure: {
-                error in
-                XCTFail("Failure block hit with error \(error)")
-                
-                expectation.fulfill()
+        UserAPI.login(with: ValidLoginEmail,
```
Comment (bryanluby) [inline Vokal-Swift.xctemplate/UserAPITests.swift]: I think the declaration was updated so the call site here needs to be updated with `login(withEmail:`. Ditto for line 82 and 111 in this file.

**Comment by scottferg** [comment]
I miss you guys 😘😘😘😘

**Comment by bryanluby** [comment]
@scottferg lets do another extralife

**Review by bryanluby — APPROVED** [review]
LGTM

**Pull request merged**